### ISSUE TITLE
evp_test: Add the missing check after calling OPENSSL_malloc

### ISSUE THREAD

**Issue by JiangJias** [openssl/openssl#17564]
The OPENSSL_zalloc() could return NULL pointer if fails.
Add the check for it does make sense, like how digest_test_init() deals with.
```
diff --git a/test/evp_test.c b/test/evp_test.c
index 6ae862b044..a97967ae04 100644
--- a/test/evp_test.c
+++ b/test/evp_test.c
@@ -574,7 +574,9 @@ static int cipher_test_init(EVP_TEST *t, const char *alg)
     }
     ERR_clear_last_mark();

-    cdat = OPENSSL_zalloc(sizeof(*cdat));
+    if (!TEST_ptr(cdat = OPENSSL_zalloc(sizeof(*cdat)))
+        return 0;
+
     cdat->cipher = cipher;
     cdat->fetched_cipher = fetched_cipher;
     cdat->enc = -1;
@@ -1175,7 +1177,9 @@ static int mac_test_init(EVP_TEST *t, const char *alg)
             return 0;
     }

-    mdat = OPENSSL_zalloc(sizeof(*mdat));
+    if (!TEST_ptr(mdat = OPENSSL_zalloc(sizeof(*mdat)))
+        return 0;
+
     mdat->type = type;
     mdat->mac_name = OPENSSL_strdup(alg);
     mdat->mac = mac;
```

**Comment by t8m** [comment]
Would you like to submit a PR with the change?

**Comment by t8m** [comment]
Fixed by #17571
